nixos/tests/end-to-end.nix

```diff
@@ -31,69 +32,79 @@ nixosTest {
-    kairos.wait_for_open_port(80)
-
-    client.wait_for_unit ("multi-user.target")
-
-    kairos.succeed("casper-client get-node-status --node-address http://localhost:11101")
-
-    # Tx Payload
-    #   nonce = 0
-    #   deposit:
-    #     amount = 1000
-    #
-    deposit_payload = "3009020100a004020203e8"
-    deposit_request = { "public_key": "cafebabe", "payload": deposit_payload, "signature": "deadbeef" }
-    # REST API
-    client.succeed("curl --fail-with-body -X POST http://kairos/api/v1/deposit -H 'Content-Type: application/json' -d '{}'".format(json.dumps(deposit_request)))
-
-    # Tx Payload
-    #   nonce = 0
-    #   transfer:
-    #     recipient = deadbabe
-    #     amount = 1000
-    #
-    transfer_payload = "300f020100a10a0404deadbabe020203e8"
-    transfer_request = { "public_key": "cafebabe", "payload": transfer_payload, "signature": "deadbeef" }
-    client.succeed("curl --fail-with-body -X POST http://kairos/api/v1/transfer -H 'Content-Type: application/json' -d '{}'".format(json.dumps(transfer_request)))
-
-    # Tx Payload
-    #   nonce = 0
-    #   withdrawal:
-    #     amount = 1000
-    #
-    withdraw_payload = "3009020100a204020203e8"
-    withdraw_request = { "public_key": "deadbabe", "payload": withdraw_payload, "signature": "deadbeef" }
-    client.succeed("curl --fail-with-body -X POST http://kairos/api/v1/withdraw -H 'Content-Type: application/json' -d '{}'".format(json.dumps(withdraw_request)))
-
-    # CLI with ed25519
-    cli_output = client.succeed("kairos-cli --kairos-server-address http://kairos deposit --amount 1000 --private-key ${testResources}/ed25519/secret_key.pem")
-    assert "ok\n" in cli_output
-
-    cli_output = client.succeed("kairos-cli transfer --recipient '01a26419a7d82b2263deaedea32d35eee8ae1c850bd477f62a82939f06e80df356' --amount 1000 --private-key ${testResources}/ed25519/secret_key.pem")
-    assert "ok\n" in cli_output
-
-    cli_output = client.succeed("kairos-cli withdraw --amount 1000 --private-key ${testResources}/ed25519/secret_key.pem")
-    assert "ok\n" in cli_output
-
-    # CLI with secp256k1
-    cli_output = client.succeed("kairos-cli --kairos-server-address http://kairos deposit --amount 1000 --private-key ${testResources}/secp256k1/secret_key.pem")
-    assert "ok\n" in cli_output
-
-    cli_output = client.succeed("kairos-cli transfer --recipient '01a26419a7d82b2263deaedea32d35eee8ae1c850bd477f62a82939f06e80df356' --amount 1000 --private-key ${testResources}/secp256k1/secret_key.pem")
-    assert "ok\n" in cli_output
-
-    cli_output = client.succeed("kairos-cli withdraw --amount 1000 --private-key ${testResources}/secp256k1/secret_key.pem")
-    assert "ok\n" in cli_output
-  '';
+  testScript =
+    let
+      cctlUsersDirectory = "/var/lib/cctl/assets/users";
+      # port see: https://github.com/casper-network/cctl/blob/745155d080934c409d98266f912b8fd2b7e28a00/utils/constants.sh#L13
+      casperNodeAddress = "http://localhost:11101";
```

Comment:
**Comment:** I was surprised to see port `11100` at above URL, but I found that [CCTL uses base port concatenated with node ID](https://github.com/casper-network/cctl/blob/ec014c13781d779738922dd1dd283435b04a1740/utils/infra.sh#L211) (starting at 1).